infra/build/build_status/update_build_status.py

```diff
@@ -0,0 +1,295 @@
+  badge = 'building'
+  # last_coverage_build_successful is False if there was an unsuccessful build
+  # and None if the target does not support coverage (e.g. Python or Java
+  # targets).
+  if last_coverage_build_successful is False:
+    badge = 'coverage_failing'
+  if not last_build_successful:
+    badge = 'failing'
+
+  print(f'[badge] {project}: {badge}')
+
+  for extension in BADGE_IMAGE_TYPES:
+    badge_name = f'{badge}.{extension}'
+
+    # Copy blob from badge_images/badge_name to badges/project/
+    blob_name = f'{BADGE_DIR}/{badge_name}'
+
+    destination_blob_name = f'{DESTINATION_BADGE_DIR}/{project}.{extension}'
+
+    status_bucket = get_storage_client().get_bucket(STATUS_BUCKET)
+    badge_blob = status_bucket.blob(blob_name)
+    status_bucket.copy_blob(badge_blob,
+                            status_bucket,
+                            new_name=destination_blob_name)
+
+
+def upload_log(build_id):
+  """Upload log file to GCS."""
+  status_bucket = get_storage_client().get_bucket(STATUS_BUCKET)
+  gcb_bucket = get_storage_client().get_bucket(build_project.GCB_LOGS_BUCKET)
+  log_name = f'log-{build_id}.txt'
+  log = gcb_bucket.blob(log_name)
+  dest_log = status_bucket.blob(log_name)
+
+  if not log.exists():
+    print('Failed to find build log', log_name, file=sys.stderr)
+    return False
+
+  if dest_log.exists():
+    return True
+
+  gcb_bucket.copy_blob(log, status_bucket)
+  return True
+
+
+def load_status_from_gcs(filename):
+  """Load statuses from bucket."""
+  status_bucket = get_storage_client().get_bucket(STATUS_BUCKET)
+  status = json.loads(status_bucket.blob(filename).download_as_string())
+  result = {}
+
+  for project in status['projects']:
+    if project['history']:
+      result[project['name']] = project['history'][0]['success']
+
+  return result
+
+
+def update_badges():
+  """Update badges."""
+  project_build_statuses = load_status_from_gcs(FUZZING_STATUS_FILENAME)
+  coverage_build_statuses = load_status_from_gcs(COVERAGE_STATUS_FILENAME)
+
+  with concurrent.futures.ThreadPoolExecutor(max_workers=32) as executor:
+    futures = []
+    with ndb.Client().context():
+      for project in Project.query():
+        if project.name not in project_build_statuses:
+          continue
+        # Certain projects (e.g. JVM and Python) do not have any coverage
+        # builds, but should still receive a badge.
+        coverage_build_status = None
+        if project.name in coverage_build_statuses:
+          coverage_build_status = coverage_build_statuses[project.name]
+
+        futures.append(
+            executor.submit(update_build_badges, project.name,
+                            project_build_statuses[project.name],
+                            coverage_build_status))
+    concurrent.futures.wait(futures)
+
+
+def main():
+  """Entry point for cloudbuild"""
+  fuzz_tag = build_project.FUZZING_BUILD_TYPE
+  fuzz_status_filename = FUZZING_STATUS_FILENAME
+
+  with ndb.Client().context():
```

Comment:
nit: just have a single ndb.Client().context() in main:

```python
if __name__ == '__main__':
  with ndb.Client.context():
     ...
```

We can also clean up main like so:


```python

configs = (
  (build_project.FUZZING_BUILD_TYPE, FUZZING_STATUS_FILENAME),
  (build_and_run_coverage.COVERAGE_BUILD_TYPE, COVERAGE_STATUS_FILENAME),
)

for tag, filename in configs:
  update_build_status(...)
```
  
And remove the remaining instance of ndb.Client().context() in `update_badges`.